CODE-GUIDELINES.md

```diff
@@ -37,3 +37,59 @@ test('...', () => {
   vi.mocked(window.windowMethod).mockResolvedValue('a string');
 });
 ```
+
+### Mock before tests, not during tests
+
+When you mock functions for some tests, it is recommended to define the mocks from the `beforeEach` callback inside a `describe`,
+so the environment created by the mocks is clearly defined in the `describe`, and the `test` is only executing the tests:
+
+You also need to reset (or restore) the mocks from the `afterEach` callback, so these mocks are not used for other tests.
+
+Do not write:
+
+```ts
+vi.mock('node:fs');
+
+test('first behaviour', () => {
+  vi.mocked(fs.existsSync).mockReturnValue(true);
+  codeToTest();
+  expect(...);
+});
+
+test('second behaviour', () => {
+  vi.mocked(fs.existsSync).mockReturnValue(true);
+  otherCodeToTest();
+  expect(...);
+});
```

Comment:
I would argue that the following is better

```ts
import { existsSync } from 'node:fs';

vi.mock('node:fs', () => ({
  existsSync: vi.fn(),
}));

beforeEach(() => {
  vi.mocked(existsSync).mockReturnValue(true);
});

test('check something', () => {
  // ...call method using existsSync
   
  expect(existsSync).toHaveBeenCalledWith('/path/');
})
````

Reply:
>   vi.resetAllMocks();

This will be problematic if you have defined spies, see discussion on https://github.com/podman-desktop/podman-desktop/pull/10541